### ISSUE TITLE
cifssrvtop fails on SmartOS

### ISSUE THREAD

**Issue by chrisridd** [richardelling/tools#8]
On joyent_20160804T173241Z, there are no probes matching ReadRaw/WriteRaw, so it looks like the -Z flag needs to be passed to dtrace.

However a more significant problem is that there is no data output at all, just the header which refreshes at the appropriate interval. e.g.

```
2016 Sep  8 20:40:39, load: 0.28, read: 0        KB, write: 0        KB
Client          CIFSOPS   Reads  Writes   Rd_bw   Wr_bw    Rd_t    Wr_t  Align%

```

**Comment by richardelling** [comment]
thanks for the report... very likely we need to update this to match the more modern SMB implementation.

This brings up an interesting question: does Joyent put in-kernel SMB into SmartOS? In other words, the symptoms you describe also match those expected if the kernel module is not loaded.

**Comment by chrisridd** [comment]
Yes they do include in-kernel SMB in SmartOS. modinfo's output includes smbsrv, and with the service enabled in a non-global zone I'm successfully accessing files from a Mac.

Running cifssrvtop in the global zone fails as I described, but running it in the same non-global zone with smb/server enabled fails differently:

```
dtrace: error on enabled probe ID 3 (ID 73154: profile:::tick-1sec): invalid kernel access in action #1 at DIF offset 20
```

and there's no other output at all.
